### PR TITLE
#44 - Test environment

### PR DESCRIPTION
**Pull request by Blusia**
The following draft files created based on files from the Toby repo and environment/dev from Keating repo.

environment/prod/app
- Dockerfile,
- entrypoint.sh
- nginx.conf,
- php.ini,
- php-fpm.conf,
- supervisord.conf

environment/prod/deployment/beta
- .env.beta.encrypted 
- docker-compose.beta.yml,
- Makefile

environment/prod/deployment/scripts
- post-deploy-actions.sh
- version.sh

.github/workflows
- deploy-to-beta-manually.yml > change to deploy-to-beta-by-push.yml

keating
- .dockerignore

### CODE REVIEW

File: .github/workflows/deploy-to-beta-manually.yml [hunk 1]
```diff
@@ -0,0 +1,95 @@
+name: Deploy to beta manually
+
+concurrency:
+  group: deploy-beta
+  cancel-in-progress: true
+
+on:
+  workflow_dispatch:
```
Comment (mtracz) [inline .github/workflows/deploy-to-beta-manually.yml:8]: Regarding to the issue https://github.com/blumilksoftware/keating/issues/44,, workflow should be triggered on merge to the main branch.

See
https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#running-your-workflow-only-when-a-push-to-specific-branches-occurs

**Review by kamilpiech97 — DISMISSED** [review]

**Review by krzysztofrewak — COMMENTED** [review]
"test environment prod draft files" means a lot. Could you be more precise with PR titles?

File: environment/prod/deployment/beta/Makefile [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+DOCKER_EXEC_SCRIPT = docker compose --file ${DOCKER_COMPOSE_FILENAME} exec --workdir /application/environment/prod/deployment/scripts ${DOCKER_COMPOSE_APP_CONTAINER} bash
+DOCKER_EXEC_ARTISAN_COMMAND = docker compose --file ${DOCKER_COMPOSE_FILENAME} exec ${DOCKER_COMPOSE_APP_CONTAINER} php artisan
+
+BETA_DOCKER_IMAGE = ghcr.io/blumilksoftware/keating:beta
+
+CURRENT_DIR = $(shell pwd)
+
+beta-artisan:
+	@echo "Running: php artisan ${ARTISAN_ARGS}" && \
+	${DOCKER_EXEC_ARTISAN_COMMAND} ${ARTISAN_ARGS}
```
Comment (mtracz) [inline environment/prod/deployment/beta/Makefile:16]: Is it needed for now? I don't think so.

File: environment/prod/deployment/beta/Makefile [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+		&& php artisan env:decrypt --env beta --key ${BETA_ENV_KEY} \
+		&& cp .env.beta /envs/.env.beta \
+		&& cp .env.beta /envs/.env"
+
+DEPLOYMENT_DATETIME = $(shell TZ=Europe/Warsaw date --rfc-3339=seconds)
+
+create-deployment-file:
+	@echo "\
+	DEPLOY_DATE='${DEPLOYMENT_DATETIME}'\
+	" > .deployment
```
Comment (kamilpiech97) [inline environment/prod/deployment/beta/Makefile:40]: @mtracz should we use data from `.deployment` file in app (footer info about deployment date)?
If not, I would remove that step from deployment - I think.

Comment (mtracz) [reply environment/prod/deployment/beta/Makefile:40]: I don't know if it is needed. Maybe for debug purposes.

File: environment/prod/deployment/beta/Makefile [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+DOCKER_EXEC_SCRIPT = docker compose --file ${DOCKER_COMPOSE_FILENAME} exec --workdir /application/environment/prod/deployment/scripts ${DOCKER_COMPOSE_APP_CONTAINER} bash
+DOCKER_EXEC_ARTISAN_COMMAND = docker compose --file ${DOCKER_COMPOSE_FILENAME} exec ${DOCKER_COMPOSE_APP_CONTAINER} php artisan
+
+BETA_DOCKER_IMAGE = ghcr.io/blumilksoftware/keating:beta
+
+CURRENT_DIR = $(shell pwd)
+
+beta-deploy: decrypt-beta-env create-deployment-file
+	@docker compose --file ${DOCKER_COMPOSE_FILENAME} pull && \
+	@docker compose --file ${DOCKER_COMPOSE_FILENAME} up --detach && \
```
Comment (mtracz) [inline environment/prod/deployment/beta/Makefile:16]:
```suggestion
	docker compose --file ${DOCKER_COMPOSE_FILENAME} up --detach && \
```

Only first line can contain `@` in multiline command.
```The @ symbol is commonly seen at the beginning of an action lines and means that the action line itself is not be be echoed on the screen as it is executed.```

**Review by kamilpiech97 — APPROVED** [review]

**Pull request merged**